fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/adapters/EventExpandableListViewAdapter.java

```diff
@@ -232,25 +236,63 @@ public void addEvent() {
    */
   @Override
   public View getChildView(
-      int groupPosition,
-      int childPosition,
-      boolean isLastChild,
-      View convertView,
-      ViewGroup parent) {
+          int groupPosition,
+          int childPosition,
+          boolean isLastChild,
+          View convertView,
+          ViewGroup parent) {
 
     // TODO : For the moment, events are displayed the same if user is attendee or organizer,
     // in the future it could be nice to have a pencil icon to allow organizer to modify an event
 
     LayoutEventBinding binding;
-    Event event = ((Event) getChild(groupPosition, childPosition));
+
     if (convertView == null) {
       LayoutInflater inflater = LayoutInflater.from(parent.getContext());
       binding = LayoutEventBinding.inflate(inflater, parent, false);
     } else {
       binding = DataBindingUtil.getBinding(convertView);
     }
 
-    binding.setEvent(event);
+    binding.openButton.setVisibility(View.GONE);
+    binding.reopenButton.setVisibility(View.GONE);
+    binding.scheduledButton.setVisibility(View.GONE);
+    binding.enterButton.setVisibility(View.GONE);
+    binding.closedButton.setVisibility(View.GONE);
+
+    Event event = (Event)getChild(groupPosition, childPosition);
+    if(event instanceof RollCall){
```

Comment:
Defining a new layout would not simplify the code in this class because the visibility of the buttons depend on other factors than the type of event. The idea behind was to use the current layout for all the events because the same information and buttons are most likely to appear in several types of events. The layout would contain all possible button names and we set to visible only the one that corresponds to our case. Do you think this would be a good solution or a separate layout is still better?

Reply:
I understand the issue. I think creating a method for the roll call changes can help with the readability of the method